gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemIntegrationTest.java

```diff
@@ -81,6 +82,13 @@ public class GoogleCloudStorageFileSystemIntegrationTest {
   // Name of the test object.
   protected String objectName = "gcsio-test.txt";
 
+  @Parameterized.Parameter public boolean testStorageClientImpl;
+
+  @Parameters
+  public static Iterable<Boolean> getTesStorageClientImplParameter() {
+    return List.of(false, true);
```

Comment:
Instead of true, false, why not provide the actual values (i.e. the Impl class) as parameter

Reply:
Sure. Instead of using boolean using `ClientType`. Which will make sure that these tests will run against new `ClientType` automatically.